src/components/elements/carousel/carousel.controls.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ReactNode, useCallback, useEffect, useState } from 'react'
+import styled, { css } from 'styled-components'
+import { ChevronLeft, ChevronRight } from './carousel.styles'
+import { TCarouselButtonType, TCarouselMode, TOffset } from './carousel.types'
+import useCarousel from './use-carousel'
+import { getOffsetValue } from './utils'
+import { TColor } from 'themes/theme.types'
+
+const CarouselButton = styled.button<{
+    button_type: TCarouselButtonType
+    mode: TCarouselMode
+    color?: TColor
+    offset: TOffset
+}>`
+    outline: 0;
+    cursor: pointer;
+    background-color: transparent;
+    touch-action: manipulation;
+    position: absolute;
+    z-index: 1;
+    border: 0;
+
+    :disabled {
+        opacity: 0.6;
+    }
+    path {
+        ${({ color }) =>
+            color
+                ? css`
+                      fill: ${color};
+                  `
+                : css`
+                      fill: unset;
+                  `}
+    }
+
+    ${({ button_type, mode }) => {
+        if (mode === 'horizontal') {
+            return button_type === 'next'
+                ? css<{ offset: TOffset }>`
+                      top: 50%;
+                      right: ${({ offset }) => getOffsetValue(offset)};
+                  `
+                : css<{ offset: TOffset }>`
+                      top: 50%;
+                      left: ${({ offset }) => getOffsetValue(offset)};
+                  `
+        } else {
+            return button_type === 'next'
+                ? css<{ offset: TOffset }>`
+                      bottom: ${({ offset }) => getOffsetValue(offset)};
+                      left: 50%;
+                      transform: rotate(90deg);
+                  `
+                : css<{ offset: TOffset }>`
+                      top: ${({ offset }) => getOffsetValue(offset)};
+                      left: 50%;
+                      transform: rotate(90deg);
+                  `
+        }
+    }}
```

Comment:
```suggestion
    ${({ button_type, mode }) => {
        let is_horizontal = mode === 'horizontal'
        let is_next_button = button_type === 'next'
        let horizontal_direction = is_next_button ? 'right' : 'left'
        let vertical_direction = is_next_button ? 'bottom' : 'top'

        return css<{ offset: TOffset }>`
            ${is_horizontal ? 'top' : 'left'}: 50%;
            ${!is_horizontal ? 'transform: rotate(90deg)' : ''};
            ${is_horizontal ? horizontal_direction : vertical_direction}: ${({ offset }) => getOffsetValue(offset)};
        `
    }}
```
Can we simplify this using helper variables?